### PR TITLE
[Guide] Intro & Quickstart guide

### PR DESCRIPTION
**Pull request by AbstractUmbra**
## Summary

(Duplicate of #7500 due to contributor inactivity).

This PR will add the Intro and Quickstart guides to the discord.py guide.

## Checklist

- [ ] If code changes were made then they have been tested.
    - [ ] I have updated the documentation to reflect the changes.
- [ ] This PR fixes an issue.
- [ ] This PR adds something new (e.g. new method or parameters).
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [x] This PR is **not** a code change (e.g. documentation, README, ...)

### CODE REVIEW

File: docs/guide/quickstart.rst [hunk 1]
```diff
@@ -0,0 +1,295 @@
...
+6. Your bot token will only be retrievable upon creation and looks like this:
+    .. image:: /images/discord_bot_token.png
+
+6a. Otherwise it will look like this:
+    .. image:: /images/discord_bot_token_no_copy.png
+
+.. danger::
+
+    This token is **VERY IMPORTANT** and you should treat it like the password to your bot. Keep it as secure as possible and never give this to anyone.
+    If you ever leak it, you can click the ``Regenerate`` button to forcibly invalidate all copies of the token. This will also terminate all running instances of this bot.
```
Comment (numbermaniac) [inline docs/guide/quickstart.rst:43]:
```suggestion
    If you ever leak it, you can click the ``Regenerate`` button to forcibly invalidate all copies of the token. This will also terminate all running instances of your bot.
```

The rest of this paragraph uses "you" and "your" so it seems appropriate to do the same here.

File: docs/guide/quickstart.rst [hunk 2]
```diff
@@ -0,0 +1,295 @@
...
+
+.. _guide_quickstart_client_secret:
+
+.. warning::
+
+    You may be thinking to use the ``Client Secret`` for your bot token. This is **NOT** what you are looking for, you need the bot **Token**, which is a completely different format.
```
Comment (numbermaniac) [inline docs/guide/quickstart.rst:49]:
```suggestion
    You may be thinking of using the ``Client Secret`` for your bot token. This is **NOT** what you are looking for - you need the bot **Token**, which is a completely different format.
```
Slightly better grammar.

File: docs/guide/quickstart.rst [hunk 3]
```diff
@@ -0,0 +1,295 @@
...
+.. note::
+
+    This walk-through does not cover application commands (slash commands). For a detailed walk-through of application commands, see _guide_app_commands.
+
+
+1. Create a new Python file in the folder you want to work in.
+
+    .. warning::
+
+        Do not name the file ``discord.py``, this will cause conflicts with the discord.py library. Also, do not create a sub-folder in your project named ``discord``, as that too will cause conflicts.
```
Comment (numbermaniac) [inline docs/guide/quickstart.rst:68]:
```suggestion
        Do not name the file ``discord.py``, as this will cause conflicts with the discord.py library. Also, do not create a sub-folder in your project named ``discord``, as that too will cause conflicts.
```

**Review by ShineyDev — CHANGES_REQUESTED** [review]
This guide is not an introduction to the library, but a guide to *installing* the library. You should consider changing the file name to `install.rst` or similar, and creating a true introduction such as [the django overview](https://docs.djangoproject.com/en/4.1/intro/overview/) or similar.

File: docs/guide/intro.rst [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+        $ .\.venv\Scripts\activate
+
+3. You can then use ``pip`` and ``python`` without interfering with other projects:
+
+    .. code-block:: shell
+
+        $ pip install -U discord.py  # note the lack of ``py -3 -m`` or ``python3 -m``
+        $ python your_bot.py
+
+For a more in-depth look into virtual environments, see :doc:`py:tutorial/venv`.
```
[the remark this answers is not in the document]
Comment (2br-2b) [reply docs/guide/intro.rst:98]: I personally feel like it's be nice to have this, even if it's redundant. I've been referred to the virtual environment documentation in the past, but it just feels like there's too much I'd have to learn in order to use a virtual environment, even if it is as simple as it appears to be. The guide in the commit gives a short, simple introduction anyone can use without spending a lot of time figuring out how it works.

Of course, when something goes wrong (as it inevitably will), they'll have to learn more (and this might involve them reaching out to community members), but I think that having this will be a net good. It'll get people's foot in the door and give them a chance to start learning about `venv`s.

[the remark this answers is not in the document]
Comment (AbstractUmbra) [reply docs/guide/intro.rst:102]: Sorry for the necro.

I do understand that point, and to some extent I agree. I'll leave it to the open forum of users to maybe 👍 or 👎 to let us know.
Unless there's an executive decision made on it.

[the remark this answers is not in the document]
Comment (JustinPanchula) [reply docs/guide/intro.rst:102]: I personally don't mind it, I realize I haven't contributed much to these docs but I always see the use in fully fleshed-out introductions.

Given that this is specifically called out in the "intro" section I feel we should definitely be giving as much help as we can

File: docs/guide/quickstart.rst [hunk 5]
```diff
@@ -0,0 +1,295 @@
...
+        :scale: 90 %
+
+3. Navigate over to the ``bot`` tab:
+    .. image:: /images/discord_bot_tab.png
+        :scale: 80%
+
+4. Create a new Bot user if you have not done so already:
+    .. image:: /images/discord_create_bot_user.png
+
+5. If you need these intents, scroll down and click on these check boxes:
```
Comment (IAmTomahawkx) [inline docs/guide/quickstart.rst:31]: Assuming these are beginners to discord bots, they likely won't know what intents they need. Perhaps this should link to the intents guide, once created.

File: docs/guide/quickstart.rst [hunk 6]
```diff
@@ -0,0 +1,295 @@
...
+    ``on_ready`` can and will be called multiple times throughout your bot's uptime. You should avoid doing any kind of state-management here, such as connecting and loading your database.
+
+.. code-block:: python
+
+    @bot.command()
+    async def echo(ctx: commands.Context, *, sentence: commands.clean_content):
+        await ctx.send(sentence)
+
+Here's where our commands will be defined. We use the :meth:`@bot.command() <ext.commands.Bot.command>` decorator to flag this function as a command.
+This creates a command ``!echo`` that we can type into a channel, and the bot will respond with the given sentence. A few key features:
```
Comment (IAmTomahawkx) [inline docs/guide/quickstart.rst:138]: This will not work given default intents.

File: docs/guide/quickstart.rst [hunk 7]
```diff
@@ -0,0 +1,295 @@
...
+- Did you regenerate your token?
+
+Be sure to copy the correct token from the bot tab on the Discord developer portal. A real token looks like: ::
+
+    MjM4NDk0NzU2NTIxMzc3Nzky.CunGFQ.wUILz7z6HoJzVeq6pyHPmVgQgV4
+
+My bot isn't responding to commands
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+If you ran a command, but the bot isn't responding, there can be a few reasons why.
```
Comment (IAmTomahawkx) [inline docs/guide/quickstart.rst:285]: add message content intent to this list

**Review by IAmTomahawkx — CHANGES_REQUESTED** [review]
Mostly good, a couple of newer developments in discord that should be updated though

File: docs/guide/intro.rst [hunk 8]
```diff
@@ -0,0 +1,107 @@
...
+
+        $ sudo dnf install libffi-devel libsodium-devel python3-devel
+
+    For Arch Linux systems:
+
+    .. code-block:: shell
+
+        $ pacman -Syu libsodium
+
+    For other distributions, please use your package manager to find libraries for ``libffi``, ``libsodium``, and the Python 3 development headers.
```
Comment (Rapptz) [inline docs/guide/intro.rst:63]: Remove the admonition.

File: docs/guide/intro.rst [hunk 9]
```diff
@@ -0,0 +1,107 @@
...
+On Windows systems, you can use the following command instead.
+
+.. code-block:: shell
+
+    py -3 -m pip install -U discord.py
+
+Voice Support
+~~~~~~~~~~~~~~
+
+Voice support (e.g. joining voice channels and playing music) is not supported by default, and can be installed by installing ``discord.py[voice]`` instead of ``discord.py``; ::
```
Comment (Rapptz) [inline docs/guide/intro.rst:37]:
```suggestion
Voice support (e.g. playing audio in voice channels) is not enabled by default and can be enabled by installing ``discord.py[voice]`` instead of ``discord.py``. ::
```

File: docs/guide/intro.rst [hunk 10]
```diff
@@ -0,0 +1,107 @@
...
+
+Introduction
+=============
+
+Welcome to discord.py, a library for Python to aid in creating applications for Discord.
+
+Prerequisites
+--------------
+
+To install discord.py you'll need Python version 3.8.0 or later.
```
Comment (Rapptz) [inline docs/guide/intro.rst:15]:
```suggestion
To install discord.py you'll need Python 3.8 or higher. Support for earlier versions of Python is not supported.
```

File: docs/guide/quickstart.rst [hunk 11]
```diff
@@ -0,0 +1,298 @@
...
+2. Create a new application if you have not made one already:
+    .. image:: /images/discord_create_app_button.png
+        :scale: 90 %
+
+3. Navigate over to the ``bot`` tab:
+    .. image:: /images/discord_bot_tab.png
+        :scale: 80%
+
+4. Create a new Bot user if you have not done so already:
+    .. image:: /images/discord_create_bot_user.png
```
Comment (Rapptz) [inline docs/guide/quickstart.rst:29]: This image is not found, but more importantly this step is no longer necessary. All applications now have a bot user by default.

File: docs/guide/quickstart.rst [hunk 12]
```diff
@@ -0,0 +1,298 @@
...
+.. note::
+
+    This walk-through does not cover application commands (slash commands). For a detailed walk-through of application commands, see _guide_app_commands.
+
+
+1. Create a new Python file in the folder you want to work in.
+
+    .. warning::
+
+        Do not name the file ``discord.py``, as this will cause conflicts with the discord.py library. Also, do not create a sub-folder in your project named ``discord``, as that too will cause conflicts.
```
Comment (Rapptz) [inline docs/guide/quickstart.rst:68]:
```suggestion
        Do not name the file ``discord.py``, as this will cause conflicts with the discord.py library. Likewise, do not create a sub-folder in your project named ``discord``, as that too will cause conflicts.
```

File: docs/guide/quickstart.rst [hunk 13]
```diff
@@ -0,0 +1,298 @@
...
+3. Now you can start creating your bot. The following steps will go over a simple bot line-by-line to help you understand what's happening.
+
+First, you need to import the discord.py library:
+
+.. code-block:: python3
+
+    import discord
+    from discord.ext import commands
+
+We import ``discord.ext.commands`` here as we will need it for our bot.
```
Comment (Rapptz) [inline docs/guide/quickstart.rst:85]: The manner of importing this is important, it must be `from discord.ext import commands` due to the fact it's a namespace package. There should be some verbiage saying so.

File: docs/guide/quickstart.rst [hunk 14]
```diff
@@ -0,0 +1,298 @@
...
+
+.. code-block:: python3
+
+    intents = discord.Intents.default()
+    intents.message_content = True
+    bot = commands.Bot(command_prefix="!", description="This is my awesome bot!", intents=intents)
+
+This is your bot instance. You can name the variable whatever you like but for simplicity's sake we will name it ``bot``.
```
Comment (Rapptz) [inline docs/guide/quickstart.rst:93]:
```suggestion
This is your bot instance. You can name the variable whatever you like but it is customary to name :class:`~ext.commands.Bot` instances ``bot``.
```

File: docs/guide/quickstart.rst [hunk 15]
```diff
@@ -0,0 +1,298 @@
...
+You will need to specify a ``command_prefix`` here, we use ``!``, but you can use any string, or a list of strings for multiple prefixes.
+We also add the :attr:`Intents.message_content` intent, so that our bot can read regular text messages. You can read more in the `Intents guide <_guide_intents>`_.
+
+.. note::
+
+    As of 2.0, the ``intents`` argument is required.
+    ``Intents`` are the method to specify which gateway events you wish to receive. ``Intents.default()`` means you will receive all events that are not locked behind bot verification.
+    For more information, see :ref:`intents_primer`.
```
Comment (Rapptz) [inline docs/guide/quickstart.rst:101]: This is unnecessary, the person reading this most likely is only familiar with 2.0. Past facing statements like this are out of place.

File: docs/guide/quickstart.rst [hunk 16]
```diff
@@ -0,0 +1,298 @@
...
+    import discord
+    from discord.ext import commands
+
+We import ``discord.ext.commands`` here as we will need it for our bot.
+
+.. code-block:: python3
+
+    intents = discord.Intents.default()
+    intents.message_content = True
+    bot = commands.Bot(command_prefix="!", description="This is my awesome bot!", intents=intents)
```
Comment (Rapptz) [inline docs/guide/quickstart.rst:91]:
```suggestion
    bot = commands.Bot(command_prefix='!', description='This is my awesome bot!', intents=intents)
```

We use single quotes throughout the documentation and code. That being said, generally subclassing is preferred nowadays since almost all the examples do it.

File: docs/guide/quickstart.rst [hunk 17]
```diff
@@ -0,0 +1,298 @@
...
+
+    As of 2.0, the ``intents`` argument is required.
+    ``Intents`` are the method to specify which gateway events you wish to receive. ``Intents.default()`` means you will receive all events that are not locked behind bot verification.
+    For more information, see :ref:`intents_primer`.
+
+.. code-block:: python
+
+    @bot.listen()
+    async def on_ready():
+        print(f"Ready! I am {bot.user} and my ID is {bot.user.id}")
```
Comment (Rapptz) [inline docs/guide/quickstart.rst:107]:
```suggestion
        print(f'Ready! I am {bot.user} and my ID is {bot.user.id}')
```

File: docs/guide/quickstart.rst [hunk 18]
```diff
@@ -0,0 +1,298 @@
...
+
+This is the ``ready`` event. It is called when the bot has finished loading and everything is cached.
```
Comment (Rapptz) [inline docs/guide/quickstart.rst:109]:
```suggestion
This is the :func:`on_ready` event. It is called when the bot has finished loading and everything necessary is cached.
```

File: docs/guide/quickstart.rst [hunk 19]
```diff
@@ -0,0 +1,298 @@
...
+
+    To remedy this, either switch to :meth:`@bot.listen() <ext.commands.Bot.listen>`, or add :meth:`~ext.commands.Bot.process_commands` to your ``on_message`` event:
+
+    .. code-block:: python
+
+        @bot.event
+        async def on_message(message: discord.Message):
+            print(f'Received "{message.clean_content}" from {message.author}')
+            # IMPORTANT:
+            await bot.process_commands(message)
```
Comment (Rapptz) [inline docs/guide/quickstart.rst:127]: This is just repeating verbiage in the FAQ already. Just link to it.

File: docs/guide/quickstart.rst [hunk 20]
```diff
@@ -0,0 +1,298 @@
...
+    @bot.listen()
+    async def on_ready():
+        print(f"Ready! I am {bot.user} and my ID is {bot.user.id}")
+
+This is the ``ready`` event. It is called when the bot has finished loading and everything is cached.
+We use the :meth:`@bot.listen() <ext.commands.Bot.listen>` decorator as to not override the main event.
+For a list of available events, see :ref:`discord-api-events`.
+
+.. _guide_quickstart_bot_event_warning:
+
```
Comment (Rapptz) [inline docs/guide/quickstart.rst:114]:
```suggestion
```

File: docs/guide/quickstart.rst [hunk 21]
```diff
@@ -0,0 +1,298 @@
...
+        @bot.event
+        async def on_message(message: discord.Message):
+            print(f'Received "{message.clean_content}" from {message.author}')
+            # IMPORTANT:
+            await bot.process_commands(message)
+
+.. warning::
+
+    ``on_ready`` can and will be called multiple times throughout your bot's uptime. You should avoid doing any kind of state-management here, such as connecting and loading your database.
```
Comment (Rapptz) [inline docs/guide/quickstart.rst:131]: This verbiage is also in the `on_ready` page. I suppose this one in particular is fine to repeat.
```suggestion
.. warning::

    :func:`on_ready` can and will be called multiple times throughout your bot's uptime. You should avoid doing any kind of state-management here, such as connecting and loading your database.
```

File: docs/guide/quickstart.rst [hunk 22]
```diff
@@ -0,0 +1,298 @@
...
+
+.. code-block:: python
+
+    @bot.command()
+    async def echo(ctx: commands.Context, *, sentence: commands.clean_content):
+        await ctx.send(sentence)
```
Comment (Rapptz) [inline docs/guide/quickstart.rst:137]: I don't think this is a particularly good intro command. It does too much from the get go. It's better if you have simpler converters (such as `int`) and simpler parameters. I know it's weird to say considering this uses a single parameter but you have to understand quite a lot from this simple example. 

The `commands.rst` guide actually tries to do it in a digestible order. It would be ideal to link back to it instead of trying to repeat its content.

**Review by Rapptz — CHANGES_REQUESTED** [review]

File: docs/guide/intro.rst (hunk 10 above)
Comment (Willy-C) [reply docs/guide/intro.rst:15]:
```suggestion
To install discord.py you'll need Python 3.8 or higher. Earlier versions of Python are not supported.
```
Fixed typo in Danny's suggested change

File: docs/guide/intro.rst [hunk 23]
```diff
@@ -0,0 +1,107 @@
...
+    .. code-block:: pwsh
+
+        $ .\.venv\Scripts\activate
+
+3. You can then use ``pip`` and ``python`` without interfering with other projects:
+
+    .. code-block:: shell
+
+        $ pip install -U discord.py  # note the lack of ``py -3 -m`` or ``python3 -m``
+        $ python your_bot.py
```
Comment (Gobot1234) [inline docs/guide/intro.rst:100]:
```suggestion
        $ python3 your_bot.py
```
Maybe this is just my fucked up path (on macos in a venv) but python by itself doesn't work

**Comment by tailoric** [comment]
As mentioned some time ago in the [#docs](https://discord.com/channels/336642139381301249/714516540720152728/1167922714737119253) channel on the discord server by me. 
Could it be possible to also explain, or at least mention, the command line commands of discord.py:
* `python -m discord newbot` 
* `python -m discord newcog` 

I've noticed that they are never mentioned in the documentation at all (unless I really missed them) and many people don't even know about these helpful features for setting up the base skeleton of both.
It could be nice to tell in the introduction about at least the `newbot` command since it sets up a good "boilerplate" to start from.
The `newcog` command may be better suited inside the Cogs guide but I think it would be helpful to give more awareness to both of them in the guides.

**Review by ShineyDev — CHANGES_REQUESTED** [review]
Since the content has been reviewed by several others by now I'll just restate that the files should be renamed. A `guide/intro` should not tell me how to install a library, consider renaming to `guide/install[ation]`. This also makes space for `guide/quickstart` to be renamed to `guide/intro[duction]`.

**Comment by AbstractUmbra** [comment]
> Since the content has been reviewed by several others by now I'll just restate that the files should be renamed. A `guide/intro` should not tell me how to install a library, consider renaming to `guide/install[ation]`. This also makes space for `guide/quickstart` to be renamed to `guide/intro[duction]`.

I personally agree with this and will rename these files appropriately. If there are issues we can always revert.

**Review by ShineyDev — APPROVED** [review]
LGTM.

File: docs/guide/introduction.rst [hunk 24]
```diff
@@ -0,0 +1,296 @@
...
+    .. code-block:: shell
+
+        $ git add .gitignore
+        $ git commit -m "Add .gitignore"
+        $ git status
+        # notice how there is no more ``config.py``!
+        On branch master
+        no changes to commit, working tree clean
+
+    .. note::
```
Comment (No767) [inline docs/guide/introduction.rst:260]: Honestly the note directly doesn't seem to make sense here to me. It could just be an regular sentence but i'm not so sure

File: docs/guide/installation.rst [hunk 25]
```diff
@@ -0,0 +1,103 @@
...
+
+    .. code-block:: shell
+
+        $ sudo dnf install libffi-devel libsodium-devel python3-devel
+
+.. tab:: Arch Linux
+
+    .. code-block:: shell
+
+        $ pacman -Syu libsodium
```
[the remark this answers is not in the document]
Comment (AbstractUmbra) [reply docs/guide/installation.rst:57]: Is this a necessary addition? If so, what's the name of the package to add?
(I do not use Arch nor can I test)

File: docs/guide/introduction.rst [hunk 26]
```diff
@@ -0,0 +1,296 @@
...
+
+.. code-block:: shell
+    :emphasize-lines: 1
+
+    $ python -m discord newcog CoolCog CoolBot/cogs
+    > successfully made cog at CoolBot/cogs/CoolCog.py
+
+This tool creates a new file within the passed directory with the provided name, and again this is a blanket implementation of an :ref:`extension <ext_commands_extensions>` file for use with the :meth:`~ext.commands.Bot.load_extension` method.
+
+The file imports the necessary items and defines a Cog class that you can extend, and adds the necessary ``setup`` method for the extension to function correctly.
```
[the remark this answers is not in the document]
Comment (AbstractUmbra) [reply docs/guide/introduction.rst:181]: Feels unnecessary to mention this, since it's covered further in the extension docs. This would likely only feel relevant if this guide was to be consumed from 1.x to 2.x migarators, not new people.

File: docs/guide/installation.rst (hunk 25 above)
[the remark this answers is not in the document]
Comment (No767) [reply docs/guide/installation.rst:57]: If `libffi` is marked as required for all of the other systems, I would assume that it would be the same case for Arch. The name of the package is [`libffi`](<https://archlinux.org/packages/core/x86_64/libffi/>)

File: docs/guide/installation.rst [hunk 27]
```diff
@@ -0,0 +1,103 @@
...
+Primer
+-------------
+
+On Unix systems you can run the following command to install discord.py from PyPI.
+
+.. code-block:: shell
+
+    python3 -m pip install -U discord.py
+
+On Windows systems, you can use the following command instead.
```
Comment (bijij) [inline docs/guide/installation.rst:26]:
```suggestion
On Windows, you can use the following command instead.
```

File: docs/guide/installation.rst [hunk 28]
```diff
@@ -0,0 +1,103 @@
...
+--------------
+
+To install discord.py you'll need Python version 3.8 or higher. Earlier versions of Python are not supported.
+
+.. _guide_install_primer:
+
+Primer
+-------------
+
+On Unix systems you can run the following command to install discord.py from PyPI.
```
Comment (bijij) [inline docs/guide/installation.rst:20]:
```suggestion
On Unix-based systems you can run the following command to install discord.py from PyPI.
```

File: docs/guide/introduction.rst [hunk 29]
```diff
@@ -0,0 +1,300 @@
...
+
+Moving on further, let's now create a basic command that can take further input from the person sending the message:-
+
+.. code-block:: python
+
+    @bot.command()
+    async def apples(ctx: commands.Context, amount: int) -> None:
+        await ctx.send(f"Hello, I would like {amount} apples please!")
+
+Here's where our commands will be defined. We use the :meth:`@bot.command() <ext.commands.Bot.command>` decorator to flag this function as a command.
```
Comment (bijij) [inline docs/guide/introduction.rst:122]: Where's here?

File: docs/guide/introduction.rst [hunk 30]
```diff
@@ -0,0 +1,300 @@
...
+This creates a command ``!apples`` that we can type into a channel, and the bot will respond with the given amount of applies in a predefined sentence.. A few key features:
+
+- :class:`~ext.commands.Context` refers to the command invocation context - this includes the :attr:`~ext.commands.Context.channel`, command :attr:`~ext.commands.Context.author`, :attr:`~ext.commands.Context.message` and more.
```
Comment (bijij) [inline docs/guide/introduction.rst:125]: I would reword the "this includes" section to use "such as ..." rather than "... and more".

File: docs/guide/introduction.rst [hunk 31]
```diff
@@ -0,0 +1,300 @@
...
+- Using ``int`` as a parameter type annotation here will instruct the library to attempt the given argument to an :class:`int` type. This means you'll have an integer in the command body and not a string.
+
+For more information on the commands extension and other converters, please reference :ref:`ext_commands_commands`.
+
+.. code-block:: python
+
+    TOKEN = "your bot token here"
+
+This is the final step, you put your bot token here, save and run the file and the bot will start up.
```
Comment (bijij) [inline docs/guide/introduction.rst:134]: nowhere in this do I see Bot.run explicitly being called. you'll need to add that.

File: docs/guide/introduction.rst [hunk 32]
```diff
@@ -0,0 +1,300 @@
...
+
+Now that you have your code ready, go to your terminal and ``cd`` into your project directory:
+
+.. code-block:: shell
+
+    $ cd my_bot_folder
+
+Activate the virtual environment (if you made one):
+
+.. code-block:: shell
```
Comment (bijij) [inline docs/guide/introduction.rst:197]: Again, use tabs.

File: docs/images/discord_client_secret_big_nono.png
Comment (bijij) [inline docs/images/discord_client_secret_big_nono.png:1]: big nono dot png

**Review by bijij — CHANGES_REQUESTED** [review]

File: docs/guide/installation.rst [hunk 33]
```diff
@@ -0,0 +1,103 @@
...
+other projects using the same Python version. They also allow you to install libraries that you may not have permission to install globally.
+
+To quickly get a virtual environment working in your project folder:
+
+1. Ensure you are in your project's root directory.
+
+    .. code-block:: shell
+
+        $ cd your_bot_source
+        $ python3 -m venv .venv
```
[the remark this answers is not in the document]
Comment (No767) [reply docs/guide/installation.rst:75]: I don't understand why creating a venv would be distro specific here. Could you explain your comment in more detail? We can have a note that says for windows systems to use the `py` launcher to gently remind users for that.
